backend/server.ts

```diff
@@ -0,0 +1,65 @@
+import express from 'express';
+import http from 'http';
+import { Server } from 'socket.io';
+
+const app: express.Express = express();
+const server: http.Server = http.createServer(app);
+const io: Server = new Server(server);
+
+const players: { [key: number]: Player } = {};
+
+app.use(express.static(__dirname + "/public"));
+
+app.get("/", function (req: express.Request<any>, res: express.Response<any>) {
+  res.sendFile(__dirname + "/index.html");
+});
+
+io.on("connection", function (socket: any) {
+  console.log("a user connected");
+
+  // create a new player and add it to our players object
+  players[socket.id] = {
+    rotation: 0,
+    x: Math.floor(Math.random() * 700) + 50,
+    y: Math.floor(Math.random() * 500) + 50,
+    playerId: socket.id,
+  };
+
+  // send the players object to the new player
+  socket.emit("currentPlayers", players);
+
+  // update all other players of the new player
+  socket.broadcast.emit("newPlayer", players[socket.id]);
+
+  // when a player disconnects, remove them from our players object
+  socket.on("disconnected", function () {
+    console.log("user disconnected");
+    // remove this player from our players object
+    delete players[socket.id];
+    // emit a message to all players to remove this player
+    io.emit("disconnect", socket.id);
```

Comment:
```suggestion
    io.emit("disconnected", socket.id);
```

```Error: "disconnect" is a reserved event name```と返されていたため、```"disconnected"```イベントを発火して client で受け取りプレイヤー消去を行う

Reply:
ありがとうございます〜
8cafa678864ae7bda1a0ddae2d6741a01e3b3e65 で修正しました